pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -289,30 +304,147 @@ func topologyHTDisabled(info *topology.Info) *topology.Info {
 	}
+
+	// Get system info
 	topologyInfo, err := ghwHandler.SortedTopology()
 	if err != nil {
-		return cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't obtain Topology Info from GHW snapshot: %v", err)
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't obtain Topology Info from GHW snapshot: %v", err)
 	}
+
 	htEnabled, err := ghwHandler.IsHyperthreadingEnabled()
 	if err != nil {
-		return cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't determine if Hyperthreading is enabled or not: %v", err)
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't determine if Hyperthreading is enabled or not: %v", err)
+	}
+
+	// Calculate reserved cpus.
+	reserved, topoInfo, err := ghwHandler.GetReservedCPUs(topologyInfo, reservedCPUCount, splitReservedCPUsAcrossNUMA, disableHTFlag, htEnabled)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
 	}
+
+	//Calculate offlined cpus
+	// note this takes into account the reserved cpus from the step above
+	offlined, err := ghwHandler.GetOfflinedCPUs(reserved, cpuInfo, topologyInfo, offlinedCPUCount, disableHTFlag, htEnabled)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	// Calculate isolated cpus.
+	// Note that topology info could have been modified by "GetReservedCPUS" so
+	// to properly calculate isolated CPUS we need to use the updated topology information.
+	isolated, err := ghwHandler.GetIsolatedCPUs(topoInfo.Nodes, reserved, offlined)
+	if err != nil {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, err
+	}
+
+	return reserved, isolated, offlined, nil
+}
+
+// Calculates Isolated cpuSet as the difference between all the cpus in the topology and those already choosen as reserved or offlined.
+// all cpus thar are not offlined or reserved belongs to the isolated cpuSet
+func (ghwHandler GHWHandler) GetIsolatedCPUs(topologyInfoNodes []*topology.Node, reserved, offlined cpuset.CPUSet) (cpuset.CPUSet, error) {
+	total, err := totalCPUSetFromTopology(topologyInfoNodes)
+	if err != nil {
+		return cpuset.CPUSet{}, err
+	}
+	return total.Difference(reserved.Union(offlined)), nil
+}
+
+func (ghwHandler GHWHandler) GetOfflinedCPUs(alreadyUsedCPUs cpuset.CPUSet, cpuInfo *cpu.Info, topologyInfo *topology.Info, offlinedCPUCount int, disableHTFlag bool, htEnabled bool) (cpuset.CPUSet, error) {
+	offlinedCPUs := newCPUAccumulator()
+	lpOfflined := 0
+
+	//currently HT is enabled on the system and the user wants to disable HT
+	if htEnabled && disableHTFlag {
+		for _, node := range topologyInfo.Nodes {
+			acc, err := offlinedCPUs.AddCoresWithFilter(offlinedCPUCount, node.Cores, func(index, lpID int) bool {
+				return !alreadyUsedCPUs.Contains(lpID) && filterFirstLogicalProcessorInCore(index, lpID)
```

Comment:
do we have a e2e test which checks the CPUs picked by this logic and ensures we offlined only thread sibilings?

Reply:
ok, but we don't actually check in the system that cpuID=X is or is not sibling of cpuID=Y, right? meaning, we hardcoded the correct setting in the test, computed manually. Is not verified querying some data, is it?
Doing like this is OK, we can extend later.

Reply:
Nope, we do not query any data in the tests, just compute the correct result manually.